### PR TITLE
[PCIIDEX] Properly handle unknown PNP IRPs. CORE-13944

### PR DESCRIPTION
**Pull request by ThFabba**

### CODE REVIEW

**Comment by HBelusca** [comment]
My question is why the original author put some ASSERT(FALSE); ? Do you think he expected to have a more involved implementation for these code paths?

**Comment by ThFabba** [comment]
I doubt it. I'd say it was just never triggered, so nobody ever cared to do something about it.
However even if handling for more PNP IRP types was needed later, it is clearly documented how to deal with unknown/unhandled ones. Driver Verifier tests and complains about this.

File: drivers/storage/ide/pciidex/pdo.c [hunk 1]
```diff
@@ -428,8 +428,7 @@ PciIdeXPdoPnpDispatch(
 				{
 					DPRINT1("IRP_MJ_PNP / IRP_MN_QUERY_DEVICE_RELATIONS / Unknown type 0x%lx\n",
 						Stack->Parameters.QueryDeviceRelations.Type);
-					ASSERT(FALSE);
-					Status = STATUS_NOT_SUPPORTED;
+					Status = Irp->IoStatus.Status;
```
Comment (HeisSpiter) [inline drivers/storage/ide/pciidex/pdo.c]: These lines look rather weird to me... Was Irp->IoStatus.Status initiliazed before?
Wouldn't it be better to set Status = STATUS_NOT_SUPPORTED;
Irp->IoStatus.Status = Status;?

Comment (ThFabba) [reply drivers/storage/ide/pciidex/pdo.c]: No, the PNP manager initializes it (to STATUS_NOT_SUPPORTED), and a higher-level driver may have set it to another status, which we must keep. The rule is: if you don't handle the specific minor function, don't touch IoStatus.
https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/dispatchpnp-routines says this somewhat indirectly, but Driver Verifier complains if you make any change to it.

Comment (HeisSpiter) [reply drivers/storage/ide/pciidex/pdo.c]: OK, thanks for the details :-).
LGTM then!

**Pull request merged**